drivers/serial/uart_nrfx_uarte.c

```diff
@@ -112,6 +120,12 @@ struct uarte_async_rx {
 	size_t offset;
 	uint8_t *next_buf;
 	size_t next_buf_len;
+#ifdef CONFIG_UART_NRFX_UARTE_ENHANCED_RX
+#if !defined(UARTE_HAS_FRAME_TIMEOUT)
+	uint32_t idle_cnt;
+#endif
+	k_timeout_t timeout;
```

Comment:
For `!defined(CONFIG_UART_NRFX_UARTE_ENHANCED_RX)`, a member with the same name but a different type is used. This is quite confusing. It would be good to rename one of them (it seems easier for the new one, to `k_timeout` maybe).

Reply:
changed to `timeout_us` for `!defined(CONFIG_UART_NRFX_UARTE_ENHANCED_RX)` case.